gaphor/ui/toolbox.py

```diff
@@ -31,60 +31,140 @@ class Toolbox(Gtk.ToolPalette):
         ),
     ]
 
-    __gsignals__ = {
-        "toggled": (
-            GObject.SignalFlags.RUN_FIRST,
-            None,
-            (GObject.TYPE_STRING, GObject.TYPE_INT),
-        )
-    }
+    title = _("Toolbox")
+    placement = ("left", "diagrams")
 
+    component_registry = inject("component_registry")
+    main_window = inject("main_window")
     properties = inject("properties")
 
-    def __init__(self, toolboxdef):
-        """Create a new Toolbox instance.
-
-        ToolItemGroups and ToolItems are created using the menu_factory and
-        based on the toolboxdef definition.
+    menu_xml = """
+      <ui>
+        <menubar name="mainwindow">
+          <menu action="diagram">
+            <separator/>
+            <menuitem action="reset-tool-after-create" />
+            <separator/>
+          </menu>
+        </menubar>
+      </ui>
+    """
 
-        """
-        GObject.GObject.__init__(self)
+    def __init__(self, toolbox_actions=TOOLBOX_ACTIONS):
+        self._toolbox = None
+        self._toolbox_actions = toolbox_actions
+        self.action_group = build_action_group(self)
+        self.action_group.get_action("reset-tool-after-create").set_active(
+            self.properties.get("reset-tool-after-create", True)
+        )
         self.buttons = []
         self.shortcuts = {}
-        self._construct(toolboxdef)
-
-    def toolbox_button(self, action_name, stock_id):
-        button = Gtk.ToolButton.new_from_stock(stock_id)
-        button.action_name = action_name
-        button.set_use_drag_window(True)
```

Comment:
For my understanding, how did you get this to work without using `set_use_drag_window`?

Reply:
The ToggleToolButton is using an internal button to do the heavy lifting. I put the DND functionality of that button, instead of the tool button. I assume the tool button has no (gdk) window so it's not able to receive any event.

See https://github.com/gaphor/gaphor/blob/diagram-specific-menu-items/gaphor/ui/toolbox.py#L86